import std;
import std.Error;

#aborts
struct TestError :: *Error {
    const char* get_name() const {
        return "TestError";
    }
    const char* get_message() const {
        return "error raised!";
    }
}

void main() {
    raise TestError{};
}